test/modules/EdgeHubRestartTester/src/MessageEdgeHubConnectorTest.cs

```diff
@@ -14,81 +14,103 @@ namespace EdgeHubRestartTester
 
-    class MessageEdgeHubConnectorTest : IEdgeHubConnector
+    class MessageEdgeHubConnectorTest : IEdgeHubConnectorTest, IDisposable
     {
-        readonly ModuleClient msgModuleClient;
         readonly Guid batchId;
-        readonly DateTime runExpirationTime;
-        readonly CancellationToken cancellationToken;
-        readonly DateTime edgeHubRestartedTime;
         readonly ILogger logger;
         long messageCount = 0;
+        ModuleClient msgModuleClient = null;
+        TestResultReportingClient reportClient = null;
 
         public MessageEdgeHubConnectorTest(
-            ModuleClient msgModuleClient,
             Guid batchId,
-            DateTime runExpirationTime,
-            DateTime edgeHubRestartedTime,
-            ILogger logger,
-            CancellationToken cancellationToken)
+            ILogger logger)
         {
-            this.msgModuleClient = Preconditions.CheckNotNull(msgModuleClient, nameof(msgModuleClient));
             this.batchId = batchId;
-            this.runExpirationTime = runExpirationTime;
-            this.cancellationToken = Preconditions.CheckNotNull(cancellationToken, nameof(cancellationToken));
-            this.edgeHubRestartedTime = edgeHubRestartedTime;
             this.logger = Preconditions.CheckNotNull(logger, nameof(logger));
         }
 
-        public async Task StartAsync()
+        public void Dispose() => this.msgModuleClient?.Dispose();
+
+        public async Task StartAsync(
+            DateTime runExpirationTime,
+            DateTime edgeHubRestartedTime,
+            CancellationToken cancellationToken)
         {
             (DateTime msgCompletedTime, HttpStatusCode mgsStatusCode) = await this.SendMessageAsync(
-                this.msgModuleClient,
                 Settings.Current.TrackingId,
                 this.batchId,
                 Settings.Current.MessageOutputEndpoint,
-                this.runExpirationTime,
-                this.cancellationToken);
+                runExpirationTime,
+                cancellationToken);
 
             TestResultBase msgTestResult = new EdgeHubRestartMessageResult(
-                Settings.Current.ModuleId + "." + TestOperationResultType.EdgeHubRestartMessage.ToString(),
+                this.GetSource(),
                 DateTime.UtcNow,
                 Settings.Current.TrackingId,
                 this.batchId.ToString(),
-                Interlocked.Read(ref this.messageCount).ToString(),
-                this.edgeHubRestartedTime,
+                this.messageCount.ToString(),
+                edgeHubRestartedTime,
                 msgCompletedTime,
                 mgsStatusCode);
 
-            var reportClient = new TestResultReportingClient { BaseUrl = Settings.Current.ReportingEndpointUrl.AbsoluteUri };
-            await ModuleUtil.ReportTestResultUntilSuccessAsync(
-                reportClient,
+            await ModuleUtil.ReportTestResultAsync(
+                this.GetReportClient(),
```

Comment:
remove this.GetReportClient() and this.logger here and move into the method.

Reply:
`ReportTestResultAsync()` lives in `ModuleUtil` class which shared across multiple classes, not only within `IEdgeHubConnectorTest`